src/MongoDB.Driver/Linq/Processors/Pipeline/MethodCallBinders/SelectManyBinder.cs

```diff
@@ -117,5 +117,22 @@ private static IFieldExpression GetCollectionField(Expression collectionSelector
 
             return null;
         }
+
+        private FieldExpression PrepareFieldExpression(IFieldExpression collectionField, IBsonSerializer serializer, PipelineBindingContext bindingContext)
+        {
+            // Determine whether we generate a query based on a document itself (when a `Document` type is `DocumentExpression`)
+            // or the result of previous steps (the `Document` type is `FieldExpression`).
+            var parentFieldExpression = collectionField.Document as FieldExpression;
+
+            if (parentFieldExpression == null)
+            {
+                return new FieldExpression(collectionField.FieldName, serializer);
+            }
+            else
+            {
+                var parentFieldSerializer = bindingContext.GetSerializer(parentFieldExpression.Serializer.ValueType, parentFieldExpression);
```

Comment:
Why do we have to look up the serializer? Why can't we use the parentFieldExpression.Serializer directly?

Reply:
Good question. By some reasons, I expected that `parentFieldExpression.Serializer directly` would contain a `collection` type in content. But I was wrong. I will fix it.

Reply:
Fixed.